refs sparkfabrik-innovation-team/board#3450 - DockerHub new pull limit (#2)
sparkfabrik/terraform-aws-ecr-pull-through-cache: README.md

# AWS ECR Pull Through cache module

> This module creates an ECR (Elastic Container Registry) [pull through cache role](https://docs.aws.amazon.com/AmazonECR/latest/userguide/pull-through-cache-creating-rule.html) reource and the resources necessary to use it.

Reviewer: ```suggestion
This module creates an ECR (Elastic Container Registry) [pull through cache role](https://docs.aws.amazon.com/AmazonECR/latest/userguide/pull-through-cache-creating-rule.html) resource and all the resources necessary to use it.
```

> 

Reviewer: ```suggestion

```

### Upstream credentials

The upstream repository credentials must be stored in an AWS Secrets Manager secret.

> The secret, with a dummy values, is created by the module with the same name as `upstream_registry` variable value plus an AWS prefix, update the secret with the real credentials.

Reviewer: ```suggestion
The secret, with a dummy values, is created by the module with the same name as `upstream_registry` variable value plus an AWS prefix. You need to update the secret with the real credentials. :warning: **ATTENTION:** this module ignores any changes for the secret value. The secret should be filled in manually in order to keep the secret value private.
```

> 

Reviewer: ```suggestion

```

### IAM Policy

> An IAM policy, with the mimimum permissions to pull images, is created by the module, attach this IAM policy to the resources that will use the pull through cache.

Reviewer: ```suggestion
An IAM policy, with the minimum permissions to pull images, is created by the module. You need to attach this IAM policy to the resources that will use the pull through cache.
```

> 

Reviewer: ```suggestion

```

## Pull through cache usage

> To pull Docker images, for example,  with pull through cache, you must use the following image format url:

Reviewer: ```suggestion
In the following examples you can see how to pull Docker images using the pull through cache using the right image format for the URL:
```

> - For Docker Hub official images: `docker pull AWS_ACCOUNT_ID.dkr.ecr.AWS_REGION.amazonaws.com/docker-hub/library/image_name:tag`

Reviewer: ```suggestion
- for Docker Hub official images: `docker pull AWS_ACCOUNT_ID.dkr.ecr.AWS_REGION.amazonaws.com/docker-hub/library/image_name:tag`.
```

- For all other Docker Hub images: `docker pull AWS_ACCOUNT_ID.dkr.ecr.AWS_REGION.amazonaws.com/docker-hub/repository_name/image_name:tag`

> Details about other upstream repositories (like Kubernetes, Quay, GitHub, GitLab..) see the [AWS references](https://docs.aws.amazon.com/AmazonECR/latest/userguide/pull-through-cache-working-pulling.html)

Reviewer: ```suggestion
For details about other upstream repositories (like Kubernetes, Quay, GitHub, GitLab, etc.), see the [AWS references](https://docs.aws.amazon.com/AmazonECR/latest/userguide/pull-through-cache-working-pulling.html).
```
